4.queueRange.js

```diff
@@ -35,6 +35,19 @@
 
 // Write your function here
 const range = (queue) => {
+    if(queue.isEmpty()) return 0; 
+
+    let max = queue.peek();
+    let min = queue.peek();
+
+    while(!queue.isEmpty()){
+        max=Math.max(max,queue.dequeue())
+    }
+    while(!queue.isEmpty()){
+        min=Math.min(min,queue.dequeue())
+    }
+
+    return Math.abs(min)+Math.abs(max)
 
 }
 
```

Comment:
I'm not sure why this is passing the test cases. If you're dequeing in the first while loop and not saving it, your queue will be empty in the second loop. Range should be determined by Max - Min. 

Try this again in one loop.